rust/src/lib.rs

```diff
@@ -299,31 +299,33 @@ pub fn set_bundled_plugin_directory(new_dir: impl AsRef<Path>) {
 pub fn user_directory() -> PathBuf {
     let user_dir_ptr: *mut c_char = unsafe { BNGetUserDirectory() };
     assert!(!user_dir_ptr.is_null());
-    let bn_user_dir = unsafe { BnString::from_raw(user_dir_ptr) };
-    PathBuf::from(bn_user_dir.to_string())
+    let user_dir_str = unsafe { BnString::into_string(user_dir_ptr) };
+    PathBuf::from(user_dir_str)
 }
 
 pub fn user_plugin_directory() -> Result<PathBuf, ()> {
     let s: *mut c_char = unsafe { BNGetUserPluginDirectory() };
     if s.is_null() {
         return Err(());
     }
-    Ok(PathBuf::from(unsafe { BnString::from_raw(s) }.to_string()))
+    let user_plugin_dir_str = unsafe { BnString::into_string(s) };
+    Ok(PathBuf::from(user_plugin_dir_str))
 }
 
 pub fn repositories_directory() -> Result<PathBuf, ()> {
     let s: *mut c_char = unsafe { BNGetRepositoriesDirectory() };
     if s.is_null() {
         return Err(());
     }
-    Ok(PathBuf::from(unsafe { BnString::from_raw(s) }.to_string()))
+    let repo_dir_str = unsafe { BnString::into_string(s) };
+    Ok(PathBuf::from(repo_dir_str))
 }
 
-pub fn settings_file_name() -> PathBuf {
+pub fn settings_file_path() -> PathBuf {
     let settings_file_name_ptr: *mut c_char = unsafe { BNGetSettingsFileName() };
     assert!(!settings_file_name_ptr.is_null());
-    let bn_settings_file_name = unsafe { BnString::from_raw(settings_file_name_ptr) };
-    PathBuf::from(bn_settings_file_name.to_string())
+    let settings_file_path_str = unsafe { BnString::into_string(settings_file_name_ptr) };
+    PathBuf::from(settings_file_path_str)
```

Comment:
File system paths are not required to be UTF-8 on all platforms. I don't know whether it matters for these sorts of paths, but it could be handled by going from a `BnString` to an `OsStr` then to a `PathBuf` rather than via `String`.